### ISSUE TITLE
ClosureResult.from_training_step_output will devide accumulate_grad_batches

### ISSUE THREAD

**Issue by lsy643** [Lightning-AI/lightning#13243]
## 🐛 Bug

When `accumulate_grad_batches=4`, the loss will be 4 times smaller than the `accumulate_grad_batches=1`.
This happens because the `OptimizerLoop._training_step` will use `from_training_step_output` to do a normalization. 

This will result a misleading result if I want to see the training loss if I want to see it in a callback like `on_train_batch_end` function. 

### Environment

- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @borda @rohitgr7 @carmocca @justusschock @ananthsub @ninginthecloud

**Comment by carmocca** [comment]
Hi!

Are you requesting that this normalization is removed? Or are you looking for a hook where you can interact with the un-normalized loss? For the latter case, you could use `training_step_end`

**Comment by lsy643** [comment]
@carmocca 
Thanks for you reply. I will try with `training_step_end`. 
By the way, why there is a normalization when `accumulate_grad_batches ` is used? E.g when  `accumulate_grad_batches == 4`,  I understand the actual batch size is 4 times larger, but there seems to be no reason to do a normalization. 
Would you mind give me some explanation to help me understand it?

**Comment by carmocca** [comment]
You can find a comprehensive explanation for this in this post: https://stackoverflow.com/a/62076913, particularly in the last snippet.
